### PR TITLE
feats(react-dialog): Adds dialog stories

### PR DESCRIPTION
**Pull request by bsunderhus**
Adds dialog stories for documentation purposes

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                                                          | Baseline (minified/GZIP) |                           PR |       Change |
| :------------------------------------------------------------------------------------------------------------------------- | -----------------------: | ---------------------------: | -----------: |
| <samp>react-dialog</samp> <br /> <abbr title='bundle-size/Dialog.fixture.js'>Dialog (including children components)</abbr> |         `0 B`<br />`0 B` | `87.193 kB`<br />`25.976 kB` | 🆕 New entry |

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/e0083356c184c87187405027a535af02577df869'>e0083356c184c87187405027a535af02577df869</a></sub>
<div id="bundleSizeComment9423"/>

File: packages/react-components/react-dialog/src/stories/DialogAlert.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import * as React from 'react';
+import { Dialog, DialogTrigger, DialogSurface, DialogTitle, DialogBody, DialogActions } from '@fluentui/react-dialog';
+import { Button } from '@fluentui/react-components';
+import story from './DialogAlert.md';
+
+export const Alert = () => {
```
Comment (ling1726) [inline packages/react-components/react-dialog/src/stories/DialogAlert.stories.tsx:6]:
```suggestion
export const AlertDialog = () => {
```

Let's be explicit here and conform with the `alertdialog` role

File: packages/react-components/react-dialog/src/stories/DialogBestPractices.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+- Dialog boxes consist of a header (`DialogTitle`), body (`DialogBody`), and footer (`DialogActions`).
+- Validate that people’s entries are acceptable before closing the dialog. Show an inline validation error near the field they must correct.
+- Modal dialogs should be used very sparingly—only when it’s critical that people make a choice or provide information before they can proceed. Thee dialogs are generally used for irreversible or potentially destructive tasks. They’re typically paired with an overlay without a light dismiss.
+
+### Don't
+
+- Don't use more than three buttons between `DialogActions`.
+- Dialog should not open another Dialog.
+- Dialog should no be without a focusable element inside.
```
Comment (ling1726) [inline packages/react-components/react-dialog/src/stories/DialogBestPractices.md:16]:
```suggestion
- Don't open a Dialog from a Dialog
- Don't use a Dialog with no focusable elements
```

File: packages/react-components/react-dialog/src/stories/DialogChangeFocus.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
+import * as React from 'react';
+import { Dialog, DialogTrigger, DialogSurface, DialogTitle, DialogBody, DialogActions } from '@fluentui/react-dialog';
+import { Button } from '@fluentui/react-components';
+
+export const ChangeFocus = () => {
```
Comment (ling1726) [inline packages/react-components/react-dialog/src/stories/DialogChangeFocus.stories.tsx:5]:
```suggestion
export const CustomFocusedElementOnOpen = () => {
```

nit: is this a better name ?

File: packages/react-components/react-dialog/src/stories/DialogChangeFocus.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+          </DialogActions>
+        </DialogSurface>
+      </Dialog>
+    </>
+  );
+};
+
+ChangeFocus.parameters = {
+  docs: {
+    description: {
```
Comment (ling1726) [inline packages/react-components/react-dialog/src/stories/DialogChangeFocus.stories.tsx:41]: nit: Would be worth adding a description here, just saying that changing the default focused element can be done in an effect

File: packages/react-components/react-dialog/src/stories/DialogCustomTrigger.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
+Native elements and Fluent components have first class support as children of `DialogTrigger`, so they will be injected automatically with the correct props for interactions and accessibility attributes.
+
+It is possible to use your own custom React component as a child of `DialogTrigger`. These components should use ref forwarding with [React.forwardRef](https://reactjs.org/docs/forwarding-refs.html#forwarding-refs-to-dom-components)
```
Comment (ling1726) [inline packages/react-components/react-dialog/src/stories/DialogCustomTrigger.md:3]: nit: also add that native HTML elements are also supported

File: packages/react-components/react-dialog/src/stories/DialogCustomTrigger.md [hunk 6]
```diff
@@ -0,0 +1,5 @@
+Native elements and Fluent components have first class support as children of `DialogTrigger`, so they will be injected automatically with the correct props for interactions and accessibility attributes.
```
Comment (ling1726) [inline packages/react-components/react-dialog/src/stories/DialogCustomTrigger.md:1]:
```suggestion
Native HTML elements and Fluent components have first class support as children of `DialogTrigger`, so they will be injected automatically with the correct props for interactions and accessibility attributes.
```

File: packages/react-components/react-dialog/src/stories/DialogCustomTrigger.stories.tsx [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+      Custom Trigger
+    </Button>
+  );
+});
+
+export const CustomTrigger = () => {
+  return (
+    <Dialog>
+      <DialogTrigger>
+        <CustomMenuTrigger />
```
Comment (ling1726) [inline packages/react-components/react-dialog/src/stories/DialogCustomTrigger.stories.tsx:19]:
```suggestion
        <CustomDialogTrigger />
```

File: packages/react-components/react-dialog/src/stories/DialogCustomTrigger.stories.tsx [hunk 8]
```diff
@@ -0,0 +1,45 @@
+import * as React from 'react';
+import { Button } from '@fluentui/react-components';
+import { Dialog, DialogSurface, DialogTitle, DialogBody, DialogActions, DialogTrigger } from '@fluentui/react-dialog';
+import type { DialogTriggerChildProps } from '@fluentui/react-dialog';
+import story from './DialogCustomTrigger.md';
+
+const CustomMenuTrigger = React.forwardRef<HTMLButtonElement, Partial<DialogTriggerChildProps>>((props, ref) => {
```
Comment (ling1726) [inline packages/react-components/react-dialog/src/stories/DialogCustomTrigger.stories.tsx:7]:
```suggestion
const CustomDialogTrigger = React.forwardRef<HTMLButtonElement, Partial<DialogTriggerChildProps>>((props, ref) => {
```

File: packages/react-components/react-dialog/src/stories/DialogDefault.stories.tsx [hunk 9]
```diff
@@ -0,0 +1,30 @@
+import * as React from 'react';
+import { Dialog, DialogTrigger, DialogSurface, DialogTitle, DialogBody, DialogActions } from '@fluentui/react-dialog';
+import type { DialogProps } from '@fluentui/react-dialog';
+import { Button } from '@fluentui/react-components';
+
+export const Default = (props: Partial<DialogProps>) => {
```
Comment (ling1726) [inline packages/react-components/react-dialog/src/stories/DialogDefault.stories.tsx:6]:
```suggestion
export const Default = () => {
```

We no longer use storybook controls in our docsite, and this code leaks into our codesandbox exports, so I don't think should include this in stories anymore

File: packages/react-components/react-dialog/src/stories/DialogNested.stories.tsx [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+        <DialogSurface aria-label="label">
+          <DialogTitle>Dialog title</DialogTitle>
+          <DialogActions>
+            <Dialog>
+              <DialogTrigger>
+                <Button appearance="primary">Open inner dialog</Button>
+              </DialogTrigger>
+              <DialogSurface aria-label="label">
+                <DialogTitle>Inner dialog title</DialogTitle>
+                <DialogBody>⚠️ just because you can doesn't mean you should have nested dialogs ⚠️</DialogBody>
```
Comment (ling1726) [inline packages/react-components/react-dialog/src/stories/DialogNested.stories.tsx:21]: Love this message 💖💖

File: packages/react-components/react-dialog/src/stories/DialogNoFocusableElement.stories.tsx [hunk 11]
```diff
@@ -0,0 +1,42 @@
+import * as React from 'react';
+import { Dialog, DialogTrigger, DialogSurface, DialogTitle, DialogBody } from '@fluentui/react-dialog';
+import { Button } from '@fluentui/react-components';
+
+export const NoFocusableElement = () => {
```
Comment (ling1726) [inline packages/react-components/react-dialog/src/stories/DialogNoFocusableElement.stories.tsx:5]: tbh I wouldn't even want to put this example in the docsite

File: packages/react-components/react-dialog/src/stories/DialogAlert.stories.tsx [hunk 12]
```diff
@@ -0,0 +1,37 @@
+import * as React from 'react';
```
[the remark this answers is not in the document]
Comment (bsunderhus) [reply packages/react-components/react-dialog/src/stories/DialogAlert.stories.tsx:1]: Good to know 🙈

File: packages/react-components/react-dialog/src/stories/DialogNoFocusableElement.stories.tsx (hunk 11 above)
Comment (bsunderhus) [reply packages/react-components/react-dialog/src/stories/DialogNoFocusableElement.stories.tsx:5]: The same way as `Nested Dialogs`, although it's not ideal, it's good to document edge cases.

File: packages/react-components/react-dialog/src/stories/DialogNested.stories.tsx (hunk 10 above)
Comment (ling1726) [reply packages/react-components/react-dialog/src/stories/DialogNested.stories.tsx:21]: honestly an image would be far better

image

**Review by ling1726 — APPROVED** [review]

File: packages/react-components/react-dialog/src/stories/DialogAlert.stories.tsx [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+import { Dialog, DialogTrigger, DialogSurface, DialogTitle, DialogBody, DialogActions } from '@fluentui/react-dialog';
+import { Button } from '@fluentui/react-components';
+import story from './DialogAlert.md';
+
+export const AlertDialog = () => {
+  return (
+    <>
```
Comment (layershifter) [inline packages/react-components/react-dialog/src/stories/DialogAlert.stories.tsx:8]: nit: redundant fragment

File: packages/react-components/react-dialog/src/stories/DialogChangeFocus.stories.tsx [hunk 14]
```diff
@@ -0,0 +1,45 @@
...
+export const CustomFocusedElementOnOpen = () => {
+  const buttonRef = React.useRef<HTMLButtonElement>(null);
+  const [open, setOpen] = React.useState(false);
+  React.useEffect(() => {
+    if (open && buttonRef.current) {
+      buttonRef.current.focus();
+    }
+  }, [open]);
+  return (
+    <>
```
Comment (layershifter) [inline packages/react-components/react-dialog/src/stories/DialogChangeFocus.stories.tsx:14]: Again a fragment :)

File: packages/react-components/react-dialog/src/stories/DialogControllingOpenAndClose.stories.tsx [hunk 15]
```diff
@@ -0,0 +1,39 @@
+import * as React from 'react';
+import { Dialog, DialogTrigger, DialogSurface, DialogTitle, DialogBody, DialogActions } from '@fluentui/react-dialog';
+import { Button } from '@fluentui/react-components';
+import story from './DialogControllingOpenAndClose.md';
+
+export const ControllingOpenAndClose = () => {
+  const [open, setOpen] = React.useState(false);
+  return (
+    <>
```
Comment (layershifter) [inline packages/react-components/react-dialog/src/stories/DialogControllingOpenAndClose.stories.tsx:9]: An again 🐱

File: packages/react-components/react-dialog/src/stories/DialogDefault.stories.tsx [hunk 16]
```diff
@@ -0,0 +1,29 @@
+import * as React from 'react';
+import { Dialog, DialogTrigger, DialogSurface, DialogTitle, DialogBody, DialogActions } from '@fluentui/react-dialog';
+import { Button } from '@fluentui/react-components';
+
+export const Default = () => {
+  return (
+    <>
```
Comment (layershifter) [inline packages/react-components/react-dialog/src/stories/DialogDefault.stories.tsx:7]: And a fragment again :)

File: packages/react-components/react-dialog/src/stories/index.stories.tsx [hunk 17]
```diff
@@ -0,0 +1,26 @@
...
+export { AlertDialog as Alert } from './DialogAlert.stories';
+export { Nested } from './DialogNested.stories';
+export { NoFocusableElement } from './DialogNoFocusableElement.stories';
+export { ControllingOpenAndClose } from './DialogControllingOpenAndClose.stories';
+export { CustomFocusedElementOnOpen as ChangeFocus } from './DialogChangeFocus.stories';
+export { TriggerOutsideDialog } from './DialogTriggerOutsideDialog.stories';
+export { CustomTrigger } from './DialogCustomTrigger.stories';
+
+export default {
+  title: 'Components/Dialog',
```
Comment (layershifter) [inline packages/react-components/react-dialog/src/stories/index.stories.tsx:17]: This should be under "Preview Components", right?

**Pull request merged**